feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/data/NodeChainIdRepository.kt

```diff
@@ -0,0 +1,68 @@
+package io.novafoundation.nova.feature_settings_impl.data
+
+import io.novafoundation.nova.caip.caip2.identifier.Caip2Identifier
+import io.novafoundation.nova.common.data.network.runtime.calls.GetBlockHashRequest
+import io.novafoundation.nova.common.utils.removeHexPrefix
+import io.novafoundation.nova.core.ethereum.Web3Api
+import io.novafoundation.nova.runtime.ethereum.Web3ApiFactory
+import io.novafoundation.nova.runtime.ethereum.sendSuspend
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.connection.node.connection.NodeConnection
+import io.novafoundation.nova.runtime.multiNetwork.connection.node.connection.NodeConnectionFactory
+import io.novasama.substrate_sdk_android.wsrpc.executeAsync
+import io.novasama.substrate_sdk_android.wsrpc.mappers.nonNull
+import io.novasama.substrate_sdk_android.wsrpc.mappers.pojo
+import java.math.BigInteger
+
+interface NodeChainIdRepository {
+
+    suspend fun requestChainId(): String
+}
+
+class NodeChainIdRepositoryFactory(
+    private val nodeConnectionFactory: NodeConnectionFactory,
+    private val web3ApiFactory: Web3ApiFactory
+) {
+
+    fun create(chain: Chain, nodeUrl: String): NodeChainIdRepository {
+        val nodeConnection = nodeConnectionFactory.createNodeConnection(nodeUrl)
```

Comment:
We should make sure the connection is shut down once we done validating it

Reply:
I think it will be nice to provide here coroutine scope and stop socket when coroutine scope is completed